### DISCUSSION TITLE
Bluetooth support

### DISCUSSION THREAD

**Discussion by vpmotion** [Louisvdw/dbus-serialbattery#120] [General]
Is it possible to have Bluetooth support. The reason I'm asking is that I currently have a battery pack that has a BMS built in and can only access the data over Bluetooth and not direct connection?

**Comment by Louisvdw** [comment]
Hi @vpmotion 
At the moment a serial connection over bluetooth is not supported, but there [is a ticket](https://github.com/Louisvdw/dbus-serialbattery/issues/13) to look at adding it in future. That would still only work for BMS that the driver support as bluetooth only replace the wire and does not specify any communications protocol with the BMS. 
What BMS does your battery pack use?

**Reply by vpmotion** [reply]
Hi.

I am unsure of what brand BMS the battery manufacturer uses and I have asked the question and waiting on a response but the battery and manufacturer are stated https://sterling-power.com/products/lithium-batteries?variant=40493851115709.

**Comment by Louisvdw** [comment]
That looks like the app for the Daly Smart BMS.

**Comment by pmarches** [comment]
I am using 2 LiFePo4 centertech batteries that have an internal JBD BMS but no external connectors. The provided android app uses bluetooth to communicate with the battery. I think they just slapped a UART to BLE connector and called it a day because the protocol used is the regular JBD protocol. 

I am done writing the python driver to read the values and propagate them to the victron side. However, before I publish code and confuse everyone, how feasible is it to add bluetooth support in this project instead? A quick glance at the source code shows dependency to a serial device throughout the code. Would it be better to emulate a serial device that in fact uses the bluetooth stack?
